osgar/tools/calibfish.py

```diff
@@ -0,0 +1,84 @@
+import cv2
+import glob
+from pprint import pprint
+import numpy as np
+
+PATTERN = (8,5)
+
+def capture(camera=0):
+    cap = cv2.VideoCapture(camera)
+    if not cap.isOpened():
+        return
+
+    print("backend:", cap.getBackendName())
+    cap.set(cv2.CAP_PROP_FRAME_WIDTH, 1280)
+    cap.set(cv2.CAP_PROP_FRAME_HEIGHT, 720)
+    cap.set(cv2.CAP_PROP_FOURCC, cv2.VideoWriter_fourcc(*"MJPG"))
+
+    count = 0
+    while True:
+        status, img = cap.read()
+        if status == False:
+            continue
+        gray = cv2.cvtColor(img,cv2.COLOR_BGR2GRAY)
+        gray = cv2.resize(gray, (img.shape[1]//2, img.shape[0]//2))
+        found, corners = cv2.findChessboardCorners(gray, PATTERN, cv2.CALIB_CB_ADAPTIVE_THRESH+cv2.CALIB_CB_FAST_CHECK+cv2.CALIB_CB_NORMALIZE_IMAGE)
+        img_copy = img.copy()
+        if found:
+            corners *= 2
+            cv2.drawChessboardCorners(img, PATTERN, corners, found)
+        cv2.imshow('img', img)
+        key = cv2.waitKey(1)
+        if key == 27:
+            return
+        if key == ord('s'):
+            if found:
+                filename = 'capture{:02d}.jpg'.format(count)
+                print(filename)
+                cv2.imwrite('images/'+filename, img_copy)
```

Comment:
why don't you use OSGAR logging functionality?

Reply:
Because I need to save just a specific images. That is what the key pres "s" is for. The same behavior is in lidarview.